pkg/plugins/policies/validation/schema.go

```diff
@@ -1,50 +1,24 @@
 package validation
 
 import (
-	"github.com/ghodss/yaml"
-	"github.com/xeipuuv/gojsonschema"
-	"google.golang.org/protobuf/encoding/protojson"
-	"google.golang.org/protobuf/proto"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"k8s.io/kube-openapi/pkg/validation/strfmt"
+	"k8s.io/kube-openapi/pkg/validation/validate"
 
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 	"github.com/kumahq/kuma/pkg/core/validators"
 )
 
-func ValidateSchema(message proto.Message, schema *gojsonschema.JSONLoader) error {
-	json, err := protojson.Marshal(message)
-	if err != nil {
-		return err
-	}
-
-	documentLoader := gojsonschema.NewBytesLoader(json)
-	result, err := gojsonschema.Validate(*schema, documentLoader)
-
-	if err != nil {
-		return err
-	}
-
-	if result.Valid() {
+func ValidateSchema(spec core_model.ResourceSpec, schema *spec.Schema) error {
+	res := validate.NewSchemaValidator(schema, nil, "", strfmt.Default).Validate(spec)
```

Comment:
Two things here:

1. Schema in the previous version was pre computed, here we call a builder (`NewSchemaValidator`) on every validation. Can we move `validate.NewSchemaValidator(schema, nil, "", strfmt.Default)` to [init function](https://github.com/kumahq/kuma/pull/5166/files#diff-a5ea8163f4f66597fe4ba7bb97cc9b681436db61355555868885d74e897ee9b6R37) of the generator? That way we only call `validator.Validate(spec)`
2. Can you name the variables? `nil` and `""` does not tell me anything about their meaning.

Reply:
1. This SchemaValidator is a lightweight object that uses `spec.Schema` for the actual validation. `spec.Schema` is unmarshalled only once in the init function. I don't want to make SchemaValidator global, because it's not stateless it has write-methods.
2. Done

Reply:
> I don't want to make SchemaValidator global, because it's not stateless it has write-methods.

Then I agree 👍